web_edit_user_filter/static/src/js/searchBar.esm.js

```diff
@@ -0,0 +1,151 @@
+/** @odoo-module **/
+
+import {SearchBar} from "@web/search/search_bar/search_bar";
+import {patch} from "@web/core/utils/patch";
+import {qweb} from "web.core";
+import {_t} from "@web/core/l10n/translation";
+import {useService} from "@web/core/utils/hooks";
+
+patch(SearchBar.prototype, "web_edit_user_filter.SearchBar", {
+    setup() {
+        this.notificationService = useService("notification");
+        var self = this;
+        this._super(...arguments);
+        $("body").on("click", ".o_searchview_facet", function (ev) {
+            $(ev.currentTarget).popover("dispose");
+            if ($(ev.currentTarget).hasClass("o_facet_remove")) {
+                return;
+            }
+            var facet_type = $(this).attr("data-type");
+            var facetId = $(this).attr("data-gp");
+            self._process_filters($(this), facet_type, facetId);
+        });
+    },
+
+    _process_filters($el, facet_type, facetId) {
+        var searchItems = this.env.searchModel.getSearchItems(
+            (facet) => facet.type === facet_type
+        );
+
+        var selectedFacet = searchItems.filter(
+            (f) => f.groupId == facetId && f.isActive === true
```

Comment:
```suggestion
            (f) => f.groupId == facetId && f.isActive === true && (f.type !== 'favorite' || f.facet !== undefined)
```

I would recommend adding this.
Otherwise, as @SAnnabelle pointed out, we will have an error when editing a favorite filter that was created before installing this module.

Reply:
I agree to implement the suggestion, but when I tried to reproduce the issue locally, it could not reproduce it.

I tried editing favorites thinking latest code base would introduced such errors, but while testing on latest code base I could not reproduce it.

@SAnnabelle , could you please let me know how can I reproduce it? So, I can push the required changes.

Reply:
To reproduce the error, I did this:
 1. Create a favorite filter
 2. Install the module 'web_edit_user_filter'
 3. Try to edit the favorite filter that I created
 4. Error is thrown
 
 The problem is that `facet` wasn't set when creating my filter since the module wasn't yet installed.

Reply:
Thank you for detailed response. I am able to reproduce it. With provided suggestion, user won't be able to see edit button for favorites whose facet is not available.

I have commented reason for the condition added. Should we notify the user that it can't be unpacked?

Reply:
Well, not displaying the edit button is enough IMO